src/main/java/com/szeloch/jan/parkingsystemmanagement/rest/controller/parkingclient/ParkingClientsController.java

```diff
@@ -0,0 +1,192 @@
+package com.szeloch.jan.parkingsystemmanagement.rest.controller.parkingclient;
+
+import com.szeloch.jan.parkingsystemmanagement.rest.controller.parkingticket.ParkingTicketNotFoundForVehicleException;
+import com.szeloch.jan.parkingsystemmanagement.rest.controller.parkingticket.ParkingTicketResource;
+import com.szeloch.jan.parkingsystemmanagement.rest.controller.vehicle.VehicleNotFoundException;
+import com.szeloch.jan.parkingsystemmanagement.rest.controller.vehicle.VehiclesController;
+import com.szeloch.jan.parkingsystemmanagement.rest.model.ParkingClient;
+import com.szeloch.jan.parkingsystemmanagement.rest.model.ParkingTicket;
+import com.szeloch.jan.parkingsystemmanagement.rest.model.Vehicle;
+import com.szeloch.jan.parkingsystemmanagement.rest.repository.ParkingClientsRepository;
+import com.szeloch.jan.parkingsystemmanagement.rest.repository.ParkingTicketsRepository;
+import com.szeloch.jan.parkingsystemmanagement.rest.repository.VehiclesRepository;
+import com.szeloch.jan.parkingsystemmanagement.utils.ParkingTicketsUtil;
+import org.springframework.hateoas.Resource;
+import org.springframework.hateoas.ResourceSupport;
+import org.springframework.hateoas.Resources;
+import org.springframework.hateoas.VndErrors;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.*;
+
+import java.net.URI;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping(value = "/parking-clients", produces = "application/hal+json")
```

Comment:
Mało REST-owy endpoint, masz pomysł na lepszy? Co to za ciekawy media type?

Reply:
W prawdziwej aplikacji rozbiłbym to może na jakiś przedrostek jeszcze, clients byłoby oddzielną ścieżką.

To Media Type Hypertext Application Language (HAL) - pewien standard HATEOAS, uznałem, że taki będzie właściwy.